pkg/apiserver/config/config.go

```diff
@@ -167,7 +167,7 @@ func TryLoadFromDisk() (*Config, error) {
 		}
 	}
 
-	conf := New()
+	conf := &Config{}
```

Comment:
`New()` creates a default config, why change this?

Reply:
Many fields have incorrect (or arbitrary) default values but they are not `omitempty`, and I think we should parse the YAML values into an empty structure so as not to create previously unconfigured, redundant values.

Reply:
But, for example, if there is no monitoring configuration in the configuration file, we still want to use the default endpoint `prometheus.kubesphere-monitoring-system.svc`, by changing to `&Config{}`, is that possible?

Reply:
Nope, but does this scenario really exist, where the configmap configuration doesn't have it but we need to use it? I always think that we should load the YAML configuration without modifying it too much, to ensure that the whole YAML remains unchanged after `Marshal` and `Unmarshal`.

Reply:
I think yes, https://github.com/kubesphere/kubesphere/blob/f50de9af65abfdcc37faa832f1960128139363ae/pkg/apiserver/authorization/options.go#L35 we use `RBAC` as the default authorization method without explicitly configured in kubesphere-config. I think default options are necessary, especially when developing. It would be difficult to prepare a complete configuration file before running your program for a newbie.